functions/src/lobby.ts

```diff
@@ -207,6 +219,56 @@ export const leaveLobby = functions.https.onCall((data: unknown, context): Promi
   });
 });
 
+export const applyLobbySettings = functions.https.onCall(async (data: unknown, context): Promise<void> => {
+  const auth = context.auth;
+  // no auth then you shouldn't be here
+  if (auth === undefined) {
+    throw new functions.https.HttpsError("permission-denied", "Not Signed In");
+  }
+  // validate code
+  if (!isLobbySettingsRequest(data)) {
+    throw new functions.https.HttpsError("invalid-argument", "Invalid lobby code!");
+  }
```

Comment:
you should consider destructing `data.lobbySettings` to make code less verbose
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment